### PR TITLE
[COM-1317] CML global variables

### PR DESCRIPTION
**Pull request by tkanzakic**
**Checklist**
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://impulsumvc.atlassian.net/browse/COM-1317

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
We are adding a couple of new fields to the `Page` and `Replica` content models in contentful. In order to update to this new version those fields will have to be added to the contentful project first.

### CODE REVIEW

**Review by Rogerpeke97 — APPROVED** [review]
Looks good to me 👍🏼

**Review by guilean — COMMENTED** [review]
Should we update the `definitions` of `Replica` and `Page` to include `globalVariables`?

File: packages/nextjs-contentful-renderer/src/generateMdx/generateMdx.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-import { Block, Model, breakpoints, replaceCmlPlaceholders } from '@cmpsr/cml';
+import { Block, Model, PropsValue, breakpoints, replaceCmlPlaceholders } from '@cmpsr/cml';
+import { merge } from 'lodash';
 
-export const generateMdx = async (blocks: Block[]): Promise<Model[]> => {
+export const generateMdx = async (blocks: Block[], globalVariables: PropsValue = {}): Promise<Model[]> => {
   const promises = blocks.map(async ({ models, propsValues }) => {
     const mdxModelsUpdated = models.map((mdxModel, index) =>
-      replaceCmlPlaceholders(mdxModel, propsValues[index] || {})
+      replaceCmlPlaceholders(mdxModel, merge({}, globalVariables, propsValues[index] || {}))
```
Comment (guilean) [inline packages/nextjs-contentful-renderer/src/generateMdx/generateMdx.ts:8]:
```suggestion
      replaceCmlPlaceholders(mdxModel, merge({}, globalVariables, propsValues[index]))
```

**Review by guilean — APPROVED** [review]

**Comment by tkanzakic** [comment]
> Should we update the `definitions` of `Replica` and `Page` to include `globalVariables`?

yes, in a different repo 😉

**Review by gvillo — APPROVED** [review]

**Review by wescosta — APPROVED** [review]
LGTM 💯

**Pull request merged**